core/src/network/reducers/get.rs

```diff
@@ -0,0 +1,436 @@
+use crate::{
+    action::{ActionWrapper, GetEntryKey, GetLinksKey, QueryKey},
+    network::{
+        query::{GetLinksNetworkQuery, NetworkQuery},
+        reducers::send,
+        state::NetworkState,
+    },
+    state::State,
+};
+use holochain_core_types::{crud_status::CrudStatus, error::HolochainError};
+use holochain_json_api::json::JsonString;
+use holochain_persistence_api::hash::HashString;
+use lib3h_protocol::{data_types::QueryEntryData, protocol_client::Lib3hClientProtocol};
+use std::convert::TryInto;
+
+fn reduce_get_entry_inner(
+    network_state: &mut NetworkState,
+    key: &GetEntryKey,
+) -> Result<(), HolochainError> {
+    network_state.initialized()?;
+    let query_json: JsonString = NetworkQuery::GetEntry.into();
+    send(
+        network_state,
+        Lib3hClientProtocol::QueryEntry(QueryEntryData {
+            requester_agent_id: network_state.agent_id.clone().unwrap().into(),
+            request_id: key.id.clone(),
+            space_address: network_state.dna_address.clone().unwrap(),
+            entry_address: key.address.clone(),
+            query: query_json.to_string().into_bytes(),
+        }),
+    )
+}
+
+pub fn reduce_get(
+    network_state: &mut NetworkState,
+    _root_state: &State,
+    action_wrapper: &ActionWrapper,
+) {
+    let action = action_wrapper.action();
+    let (key_type, payload) = unwrap_to!(action => crate::action::Action::Query);
+    let result = match key_type {
+        QueryKey::Entry(key) => reduce_get_entry_inner(network_state, key)
+            .map(|_| None)
+            .unwrap_or_else(|e| Some(Err(e))),
+        QueryKey::Links(key) => {
+            let (crud_status, query) = unwrap_to!(payload => crate::action::QueryPayload::Links);
+            reduce_get_links_inner(network_state, &key, &query, crud_status)
+                .map(|_| None)
+                .unwrap_or_else(|e| Some(Err(e)))
+        }
+    };
+    network_state.get_results.insert(key_type.clone(), result);
+}
+
+pub fn reduce_get_timeout(
+    network_state: &mut NetworkState,
+    _root_state: &State,
+    action_wrapper: &ActionWrapper,
+) {
+    let action = action_wrapper.action();
+    let key = unwrap_to!(action => crate::action::Action::QueryTimeout);
+    if network_state.get_results.get(&key).is_none() {
+        return;
+    }
+
+    if network_state.get_results.get(key).unwrap().is_none() {
+        network_state
+            .get_results
+            .insert(key.clone(), Some(Err(HolochainError::Timeout)));
+    }
+}
+
+fn reduce_get_links_inner(
+    network_state: &mut NetworkState,
+    key: &GetLinksKey,
+    get_links_query: &GetLinksNetworkQuery,
+    crud_status: &Option<CrudStatus>,
+) -> Result<(), HolochainError> {
+    network_state.initialized()?;
+    let query_json: JsonString = NetworkQuery::GetLinks(
```

Comment:
Ah, this is why we still have those separate inner reducers. The `NetworkQuery` gets created here. I've overlooked this above in the action creator, but I thought we could move that over there. So create this `NetworkQuery` there and pass it over to here as the parameter of the `Query` action.